GitUI/GitUIExtensions.cs

```diff
@@ -113,11 +113,9 @@ void OpenWithDiffTool()
                         : diffOfConflict;
                 }
 
-                if (file.IsSubmodule
-                    && file.GetSubmoduleStatusAsync() is Task<GitSubmoduleStatus> task)
+                if (file.IsSubmodule)
                 {
-                    // Patch already evaluated
-                    var status = ThreadHelper.JoinableTaskFactory.Run(() => task);
+                    var status = ThreadHelper.JoinableTaskFactory.Run(file.GetSubmoduleStatusAsync!);
```

Comment:
1. Rant: The syntax of `!` is weird in this case. It applies to the _return value_ of the function `file.GetSubmoduleStatusAsync`, which is _not called_ here - but later by `Run`.
2. Are we sure that `file._submoduleStatus` is never `null`?

--- 

3. Does `JoinAsync()` always return the same `Task` instance? `GetSubmoduleStatusAsync()` is called pretty often.

Reply:
1. This is compiler info to ignore that the value may be null, it does not matter that function is not called
2. GetSubmoduleStatus is _currently_ always set. Anyway, this is a fix for the current solution that hangs if status is not yet ready
3. Do not know, I assume so